gufe/chemicalsystem.py

```diff
@@ -10,7 +9,7 @@
 from .components import Component
 
 
-class ChemicalSystem(abc.Mapping, GufeTokenizable):
```

Comment:
@dotsdl was there a reason you removed the mapping-like behaviour here?

Reply:
It wasn't clear to me that we were really using it, and it was messing with class resolution for our call to `super().__hash__()` in `__hash__`. I've re-added it and specified which parent class to `super` on in 71d83c9.

Reply:
Switched the order to avoid having to do a `super(abc.Mapping, self).__hash__()`, and instead we keep `super().__hash__()` here.